src/aspire/basis/ffb_2d.py

```diff
@@ -65,13 +66,24 @@ def _precomp(self):
         radial = np.zeros(shape=(n_r, np.sum(self.k_max)))
         ind_radial = 0
         for ell in range(0, self.ell_max + 1):
+            jv_ell = []
             for k in range(1, self.k_max[ell] + 1):
-                radial[:, ind_radial] = jv(ell, self.r0[k - 1, ell] * r / self.kcut)
+                jv_ell_k = jv(ell, self.r0[k - 1, ell] * r / self.kcut)
                 # NOTE: We need to remove the factor due to the discretization here
                 # since it is already included in our quadrature weights
                 nrm = 1 / (np.sqrt(np.prod(self.sz))) * self.basis_norm_2d(ell, k)
-                radial[:, ind_radial] /= nrm
+                jv_ell_k_norm = jv_ell_k / nrm
+                radial[:, ind_radial] = jv_ell_k_norm
+                # CTF filters use different normalized factors
```

Comment:
I don't quite understand. Why should the normalization be different here? Are you sure this is not just a bug in the `filter_to_fb_mat` code?

Reply:
There filer_to_fb_mat is almost line to line converting from the original Matlab as below, 
  https://github.com/PrincetonUniversity/aspire/blob/master/projections/radial_filter_to_fb_mat.m
I compared the results and they are consistent when I did the Cov2D.

Reply:
Well there might have been a bug in the Matlab code as well. The question remains, why should the normalization be different? Do we get the wrong result when we use the same normalization?

Reply:
Let's make a separate issue for the normalization and have this PR focus on optimizing the code.

Since the required values already exist in `basis.radial`, let's use those in `filter_to_fb_mat` by multiplying with the right constant to get the desired normalization factor. There's no need to store them in a separate `jv_norm` attribute.

Reply:
Also, here we're relying on the fact that we use the same number of radial points as in `basis._precomp`. That will not necessarily be the case always. To make this dependence explicit, we either need to store that value `n_r` as an attribute of `basis` that can then be retrieved by `filter_to_fb_mat` (with a line `n_k = fbasis.n_r`) or just rely on the fact that `n_k = fbasis.r0.shape[0]`, but I find this a less clear.

Reply:
I checked again. the Matlab version has the implementations as the Python version. In the paper of FFB basis, the normalization factor is `N_lk = 1/(c*sqrt(pi) *J_(l+1)(R_lk))`,  with c=0.5 for radius cutoff, and J for the Bessel function. This normalization factor used for filter expanding, `N_lk = 1/(c *J_(l+1)(R_lk)*sqrt(1/2)`, does not include the sqrt(pi) but addition sqrt(1/2) as shown below: 
https://github.com/PrincetonUniversity/aspire/blob/f152e8efd9856f75c294f453ef7846bd3a684410/projections/radial_filter_to_fb_mat.m#L43

In the implementation of FFB basis,  N_lk = 1/(c*sqrt(pi) *J_(l+1)(R_lk)*sqrt(1/2)), with additional sqrt(1/2) to the paper as shown
https://github.com/PrincetonUniversity/aspire/blob/master/basis/ffb_basis.m#L275
https://github.com/PrincetonUniversity/aspire/blob/master/basis/ffb_basis.m#L182

The outputs from the Cob2d_simulation example with current implementation of different normalization factors are list as below:

> Deviation of the noisy mean estimate: 0.008807039379139734
> 2020/08/20 15:41:03 Deviation of the noisy covariance estimate: 0.032321671019596605
> 2020/08/20 15:41:03 Estimated images normalized RMSE: 0.22725983789333037

Using the same normalization factor from Filter expansion, the outputs are 

> Deviation of the noisy mean estimate: 0.0088070393791396
> 2020/08/28 22:11:10 Deviation of the noisy covariance estimate: 0.22049860794751824
> 2020/08/28` 22:11:10 Estimated images normalized RMSE: 3.713621537096643

Using the same normalization factor from FFB basis, the outputs are 

> 2020/08/28 23:10:40 Deviation of the noisy mean estimate: 5.267011983601262
> 2020/08/28 23:10:40 Deviation of the noisy covariance estimate: 16.983344615954522
> 2020/08/28 23:10:40 Estimated images normalized RMSE: 3.399225365537338

In summary, the current implementation with different normalization factors have lower errors.  

Regarding performance improvements, the `filter_to_fb_mat function` used to take 5.8%(1426ms) wall time of the cov2d_simulation example on my 8 core laptop. Now it reduces to 0.2%(66ms) using precomputed values for Bessel functions on my 8 core laptop.

Reply:
Creates new issue #250 for understanding the different normalization factors.

Reply:
There shouldn't be any need for reshaping, just indexing and renormalizing of `_precomp['radial']` here. Having the same data represented in two different ways is not simpler. Please get rid of `jv_norm` and use `_precomp['radial']` directly.

Reply:
`jv_norm` was removed in new commit  190914d.